versions/v5/functions/display-logic.php

```diff
@@ -0,0 +1,231 @@
+		<?php if ( $thumbnail ): ?>
+		<div class="fp-feature-img-wrap">
+			<?php echo $thumbnail; ?>
+
+			<?php if ( $show_vertical && $vertical ): ?>
+			<span class="fp-vertical">
+				<?php echo $vertical; ?>
+			</span>
+			<?php endif; ?>
+		</div>
+		<?php endif; ?>
+	</a>
+	<div class="fp-feature-text-wrap">
+		<<?php echo $heading; ?> class="fp-feature-title">
+			<a class="fp-feature-link" href="<?php echo get_permalink( $story->ID ); ?>">
+				<?php echo $title; ?>
+			</a>
+		</<?php echo $heading; ?>>
+		<div class="fp-feature-description">
+			<?php echo $description; ?>
+		</div>
+	</div>
+</article>
+<?php 	return ob_get_clean();
+}
+
+
+/**
+ * Displays a single Today article on the front page.
+ **/
+function display_front_page_today_story( $article ) {
+	$url = $article->get_link();
+	$title = $article->get_title();
+	$publish_date = $article->get_date('m/d');
+
+	ob_start();
+?>
+<article class="fp-today-feed-item">
+	<a class="fp-today-item-link" href="<?php echo $url; ?>">
+		<div class="publish-date"><?php echo $publish_date; ?></div>
+		<?php echo $title; ?>
+	</a>
+</article>
+<?php 	return ob_get_clean();
+}
+
+
+/**
+ * Displays a single event item on the front page.
+ **/
+function display_front_page_event( $event ) {
+	$start = strtotime( $event['starts'] );
+	$description = substr( strip_tags( $event['description'] ), 0, 250 );
+	if ( strlen( $description ) == 250 ) {
+		$description .= '...';
+	}
+
+	ob_start();
+?>
+<div class="fp-event">
+	<div class="fp-event-when">
+		<span class="fp-event-day"><?php echo date( 'D', $start ); ?></span>
+		<span class="fp-event-date"><?php echo date( 'd', $start ); ?></span>
+		<span class="fp-event-month"><?php echo date( 'M', $start ); ?></span>
+	</div>
+	<div class="fp-event-content">
+		<span class="fp-vertical"><?php echo $event['category']; ?></span>
+		<span class="fp-event-title">
+			<a class="fp-event-link" href="<?php echo $event['url']; ?>"><?php echo $event['title']; ?></a>
+		</span>
+		<div class="fp-event-description">
+			<?php echo $description; ?>
+		</div>
+	</div>
+</div>
+<?php 	return ob_get_clean();
+}
+
+
+/**
+ * Displays a single featured gallery on the front page.
+ **/
+function display_front_page_gallery( $gallery, $css_class='' ) {
+	if ( !$gallery ) { return false; }
+
+	$title = wptexturize( $gallery->post_title );
+
+	$vertical = get_the_category( $gallery->ID );
+	if ( $vertical ) {
+		$vertical = $vertical[0];
+		$vertical = wptexturize( $vertical->name );
+	}
+
+	$thumbnail = null;
+	if ( get_relevant_version( $gallery ) >= 5 ) {
+		// Version 5+: fetch featured image ID
+		$thumbnail_id = get_post_thumbnail_id( $gallery );
+		$thumbnail_size = 'frontpage-featured-gallery-thumbnail-3x2';
+	} else {
+		// Version 4 and prior: get the ID from the
+		// `story_frontpage_gallery_thumb` meta field
+		$thumbnail_id = intval( get_post_meta( $gallery->ID, 'story_frontpage_gallery_thumb', true ) );
```

Comment:
Same comment as from before. This would evaluate to `0`, instead of `false` in the case of a null value, giving it a different value than the `get_post_thumbnail_id` call above on line 142.